be/test/vec/data_types/from_string_test.cpp

```diff
@@ -0,0 +1,177 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "gtest/gtest_pred_impl.h"
+#include "olap/types.h" // for TypeInfo
+#include "olap/wrapper_field.h"
+#include "vec/columns/column.h"
+#include "vec/core/field.h"
+#include "vec/data_types/data_type.h"
+#include "vec/data_types/data_type_factory.hpp"
+#include "vec/data_types/serde_utils.h"
+#include "vec/io/reader_buffer.h"
+
+namespace doris::vectorized {
+
+/**
+ * This test is used to check wrapperField from_string is equal to data type from_string or not
+ *  same string feed to wrapperField and data type from_string, and check the result from
+ *  wrapperField and data type to_string is equal or not
+ */
+TEST(FromStringTest, ScalaWrapperFieldVsDataType) {
+    // arithmetic scala field types
+    {
+        std::vector<FieldType> arithmetic_scala_field_types = {
+                FieldType::OLAP_FIELD_TYPE_BOOL,       FieldType::OLAP_FIELD_TYPE_TINYINT,
+                FieldType::OLAP_FIELD_TYPE_SMALLINT,   FieldType::OLAP_FIELD_TYPE_INT,
+                FieldType::OLAP_FIELD_TYPE_FLOAT,      FieldType::OLAP_FIELD_TYPE_DOUBLE,
+                FieldType::OLAP_FIELD_TYPE_BIGINT,     FieldType::OLAP_FIELD_TYPE_LARGEINT,
+                FieldType::OLAP_FIELD_TYPE_CHAR,       FieldType::OLAP_FIELD_TYPE_VARCHAR,
+                FieldType::OLAP_FIELD_TYPE_STRING,     FieldType::OLAP_FIELD_TYPE_DECIMAL,
+                FieldType::OLAP_FIELD_TYPE_DECIMAL32,  FieldType::OLAP_FIELD_TYPE_DECIMAL64,
+                FieldType::OLAP_FIELD_TYPE_DECIMAL128I};
+
+        for (auto type : arithmetic_scala_field_types) {
+            DataTypePtr data_type_ptr;
+            if (type == FieldType::OLAP_FIELD_TYPE_DECIMAL32) {
+                data_type_ptr = DataTypeFactory::instance().create_data_type(type, 9, 0);
+            } else if (type == FieldType::OLAP_FIELD_TYPE_DECIMAL64) {
```

Comment:
test is too simple。
For decimal type, please test decimalv2, decimalv3(decimal32,decimal64,decimal128) and the precision should be 0 or 2

Reply:
and datetimev2 also has pricision, should also test different precisions.